### PR TITLE
Add new module options to HomePage

### PR DESCRIPTION
**Pull request by chosak**
This commit adds the ability to add JumboHero and InfoUnitGroup modules to the HomePage class. It copies the IUG rendering styling from the CampaignPage.

This also fixes a bug where home page InfoUnit headings weren't getting rendered properly.

## How to test this PR

To experiment with a new design:

1. Using a production dump, [edit the home page](http://localhost:8000/admin/pages/319/edit/).
2. Delete all existing Carousel content.
3. Delete and edit existing InfoUnit content as desired.
4. Add a new JumboHero organism and a new Features organism to the page.

Preview!

(Or, CFPB only can test at https://cfgov-pr-6260.internal.domain.redacted/.)

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)

### CODE REVIEW

**Review by csebianlander — APPROVED** [review]
Tested on https://cfgov-pr-6260.redacted/ -- all works great and as expected!

File: cfgov/v1/jinja2/v1/home_page.html [hunk 1]
```diff
@@ -13,7 +13,25 @@ <h1 class="u-visually-hidden">
...
-{{ carousel.render( carousel_items ) }}
+{% for block in page.content %}
+    {% if block.block_type == 'features' %}
+        <section class="wrapper
+                        wrapper__match-content
+                        block
+                        block__bg
+                        block__border
+                        block__flush-top
+                        u-mb0">
```
Comment (anselmbradford) [inline cfgov/v1/jinja2/v1/home_page.html:24]: Would `block__flush-bottom` not work in place of `u-mb0`?

Comment (Scotchester) [reply cfgov/v1/jinja2/v1/home_page.html:24]: I wrote the code he's copying from and am sure I tried that. I think the issue is that it removes the bottom border.

Comment (chosak) [reply cfgov/v1/jinja2/v1/home_page.html:24]: Thanks @Scotchester, I was just writing the same. If I replace `u-mb0` with `block__flush-bottom`, the spacing stays the same but we lose the border.

File: cfgov/v1/models/home_page.py [hunk 2]
```diff
@@ -2,49 +2,48 @@
...
+
     card_heading = models.CharField(max_length=40, null=True, blank=True)
 
     # Tab handler interface
     edit_handler = TabbedInterface([
+        ObjectList([StreamFieldPanel('content')], heading='Content'),
         ObjectList([
             InlinePanel(
-                'carousel_items', min_num=4, max_num=4, label="Carousel Item"
+                'carousel_items', max_num=4, label="Carousel Item"
```
Comment (Scotchester) [inline cfgov/v1/models/home_page.py:41]: Why remove the `min_num` here? I shudder to think at the carousel returning, but should it, it's still only built to handle exactly 4 items.

**Review by Scotchester — CHANGES_REQUESTED** [review]
The heading level issue is pretty important.

File: cfgov/v1/models/home_page.py (hunk 2 above)
Comment (chosak) [reply cfgov/v1/models/home_page.py:41]: If we have a `min_num`, it's impossible to remove the carousel. We want the minimum number of carousel items to be zero.

File: cfgov/v1/models/home_page.py [hunk 3]
```diff
@@ -162,7 +159,10 @@ def as_info_unit(self):
             'image': {
                 'upload': self.image,
             },
-            'heading': '<h3>%s</h3>' % self.title,
+            'heading': {
+                'text': self.title,
+                'level': 'h3',
+            },
```
[the remark this answers is not in the document]
Comment (chosak) [reply cfgov/v1/models/home_page.py]: Good point. I've rebased in a change that does `<h2 class="h3">`.

File: cfgov/v1/models/home_page.py (hunk 2 above)
Comment (Scotchester) [reply cfgov/v1/models/home_page.py:41]: Ah, right!

**Review by Scotchester — APPROVED** [review]

**Pull request merged**